src/discussion.cairo

```diff
@@ -0,0 +1,135 @@
+use starknet::ContractAddress;
+
+#[starknet::interface]
+trait IDiscussion<TContractState> {
+    fn add_comment(ref self: TContractState, prop_id: felt252, ipfs_hash: felt252);
+    fn get_comment(self: @TContractState, prop_id: felt252) -> Array<(ContractAddress, felt252)>;
+} 
+
+#[starknet::component]
+mod discussion {
+    use array::ArrayTrait;
+    use core::box::Box;
+    use core::serde::Serde;
+
+    use starknet::get_caller_address;
+    use starknet::ContractAddress;
+
+    use konoha::proposals::proposals as proposals_component;
+    use konoha::proposals::proposals::ProposalsImpl;
+    use konoha::traits::IERC20Dispatcher;
+    use konoha::traits::IERC20DispatcherTrait;
+    use konoha::traits::get_governance_token_address_self;
+
+    // Storage implementation entails comments and comment_count
+    // Comments is mapping of (proposal id, index) to comment
+    // While the comment_count is a mapping of proposal id to number of comments
+    #[storage]
+    struct Storage {
+        comments: LegacyMap::<(felt252, u64), Comment>,
+        comment_count: LegacyMap::<felt252, u64>
+    }
+
+    #[derive(Copy, Drop, Serde, starknet::Store)]
+    pub struct Comment {
+        user: ContractAddress,
+        ipfs_hash: felt252,
```

Comment:
How will a IPFS hash fit into felt252? It will not. You must use a different data structure, possibly ByteArray.